docs: document the Kafka v2 preview release (#5231)
n8n-io/n8n-docs: docs/integrations/builtin/app-nodes/n8n-nodes-base.kafka.md

- Send message

## Kafka node version 2 preview

{% hint style="info" %}
**Feature availability**

Version 2 of the Kafka node is a preview. Preview features may change in future releases, so avoid relying on them in production workflows.
{% endhint %}

Version 2 of the Kafka node runs on [`@confluentinc/kafka-javascript`](https://github.com/confluentinc/confluent-kafka-javascript), Confluent's supported client built on librdkafka. It replaces kafkajs, the unmaintained library that version 1 uses. The node keeps the same fields and credential as version 1. The only visible change is the new **Compression** parameter, which compresses messages you send with **GZIP**, **Snappy**, **LZ4**, or **Zstd**. If a version 1 Kafka Trigger consumes the topic, keep **Compression** set to **GZIP** or **None**: version 1 triggers can't decode the other codecs.

> Version 2 stays off during the preview. Nodes added from the nodes panel stay on version 1, and there's no version picker in the editor. To opt in, import a workflow JSON where the Kafka node has `"typeVersion": 2`. n8n verified the preview on the official n8n Docker images, on both amd64 and arm64. Version 2 includes a native compiled component, so other install methods, such as npm or custom images, aren't verified during the preview.

Reviewer: ```suggestion
Version 2 is disabled by default during the preview. Nodes added from the nodes panel stay on version 1, and there's no version picker in the editor. To opt in, import a workflow JSON where the Kafka node has `"typeVersion": 2`. n8n has verified the preview on the official n8n Docker images, on both amd64 and arm64. Version 2 includes a native compiled component, so other install methods, such as npm or custom images, aren't verified during the preview.
```
/nitpick

> Version 2 stays off during the preview. Nodes added from the nodes panel stay on version 1, and there's no version picker in the editor. To opt in, import a workflow JSON where the Kafka node has `"typeVersion": 2`. n8n verified the preview on the official n8n Docker images, on both amd64 and arm64. Version 2 includes a native compiled component, so other install methods, such as npm or custom images, aren't verified during the preview.

Reviewer: > To opt in, import a workflow JSON where the Kafka node has `"typeVersion": 2`.

Will it be obvious to most users where to find one? Do we need to include an example?
